airavata-api/airavata-api-server/src/main/java/org/apache/airavata/api/server/handler/AiravataServerHandler.java

```diff
@@ -338,6 +338,20 @@ public String addGateway(AuthzToken authzToken, Gateway gateway) throws InvalidR
             permissionType.setDescription("Write permission type");
             sharingClient.createPermissionType(permissionType);
 
+            //Create an "everyone" group for the domain
+            String groupId = "everyone@" + domain.domainId;
+            UserGroup userGroup = new UserGroup();
+            userGroup.setGroupId(groupId);
+            userGroup.setDomainId(domain.domainId);
+            userGroup.setGroupCardinality(GroupCardinality.MULTI_USER);
+            userGroup.setCreatedTime(System.currentTimeMillis());
+            userGroup.setUpdatedTime(System.currentTimeMillis());
+            userGroup.setOwnerId(authzToken.getClaimsMap().get(Constants.USER_NAME));
```

Comment:
OwnerId should be in the form "username@domainId"

Reply:
Resolved.